Limit orders UI improvements I

fairlighteth opened the pull request:
# Summary

- Adds a 'my orders' button in the Trade container on smaller desktop resolutions and smaller
- When on 'Swap' clicking the button will trigger the Activity modal
- When on 'Limit' or 'TWAP' it will scroll down to the 'Your orders' section
<img width="513" alt="Screenshot 2024-03-05 at 15 21 09" src="https://github.com/cowprotocol/cowswap/assets/31534717/7ef35118-196b-4436-9945-9b2aba578839">


https://github.com/cowprotocol/cowswap/assets/31534717/9cbd20f4-5b0e-4fe4-8bc2-de360f1b7ee0


https://github.com/cowprotocol/cowswap/assets/31534717/29f5c4a0-a511-4eb8-aec8-a5227db3ef01

alfetopito commented:
The broken build is my fault.
The fix was added only to the release.
Here I'm cherry-picking it to develop https://github.com/cowprotocol/cowswap/pull/4003

elena-zh reviewed (changes requested):
Hey @fairlighteth , great job!

Some issues I faced: 
1. It would be great to remove the button from Swap form, not connected state: it opens an empty modal
![not connected](https://github.com/cowprotocol/cowswap/assets/70885163/7006febb-ac15-4838-bcc7-10afcea2eeef)
2. (not sure), but IMO the button is hardly visible in the dark mode
3. It would be great to remove the button from TWAP/Limit orders form when 'Unlock' banner is displayed: when I press on the button, nothing happens
4. It would be great to remove the button from 'Recreate' form (also when I press on it, nothing happens)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/559c4f39-b0b4-4b07-84c0-e036ee6d2563)

Thank you!

shoom3301 commented:
<img width="229" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/d33ebefb-36d0-4bc4-8979-8f98e82e7c92">

I'm not sure, but probably the border of the button should not be black

shoom3301 commented on apps/cowswap-frontend/src/modules/trade/containers/TradeWidget/TradeWidgetForm.tsx:
```diff
@@ -26,6 +32,13 @@ import { useIsWrapOrUnwrap } from '../../hooks/useIsWrapOrUnwrap'
 import { WrapFlowActionButton } from '../WrapFlowActionButton'
 
+const scrollToMyOrders = () => {
+  const element = document.getElementById(MY_ORDERS_ID)
```
I think it's ok to use `"my-orders"` here instead of const, otherwise we should use the const in `OrdersTableContainer/index.tsx` as well

fairlighteth replied:
Addressed this by also using the `const` in the other file

shoom3301 commented on apps/cowswap-frontend/src/modules/trade/containers/TradeWidget/TradeWidgetForm.tsx:
```diff
@@ -90,6 +105,18 @@ export function TradeWidgetForm(props: TradeWidgetProps) {
+  const isSwapMode = location.pathname.includes('/swap/')
+  const toggleAccountModal = useToggleAccountModal()
+
+  const handleClick = () => {
```
```
const handleClick = useCallback(() => {
    if (isSwapMode) {
      toggleAccountModal()
    } else {
      scrollToMyOrders()
    }
  }, [isSwapMode, toggleAccountModal])
```

fairlighteth replied:
Thank you for the suggestion. Addressed now.

alfetopito commented on apps/cowswap-frontend/src/modules/trade/containers/TradeWidget/TradeWidgetForm.tsx:
```diff
@@ -90,6 +105,18 @@ export function TradeWidgetForm(props: TradeWidgetProps) {
   }, [])
 
+  const location = useLocation()
+  const isSwapMode = location.pathname.includes('/swap/')
```
I think we have better ways to detect this, no?
I searched and apparently not.
Can you create a helper hook for this, then?

Something like:

```
function useIsSwapForm():boolean {
  return useLocation().pathname.includes('/swap/')
}
```

fairlighteth replied:
Yeah didn't find a better way other than checking the current route. Addressed now.

alfetopito commented on apps/cowswap-frontend/src/modules/trade/containers/TradeWidget/TradeWidgetForm.tsx:
```diff
@@ -98,6 +125,13 @@ export function TradeWidgetForm(props: TradeWidgetProps) {
             <AccountElement isWidgetMode={isInjectedWidgetMode} pendingActivities={pendingActivity} />
           )}
+
+          {isUpToLarge && (
+            <ButtonOutlined margin={'0 16px 0 auto'} onClick={handleClick}>
+              My orders <SVG src={ICON_ORDERS} />
+            </ButtonOutlined>
+          )}
```
As Elena noticed, we should not show this on recreate/edit order.
I can help you with this.

fairlighteth replied:
@alfetopito This has been addressed now

fairlighteth commented:
Addressed all the points now. Right now the 'My orders' button is only shown on these conditions:
- NOT in recreate order mode
- Swap Mode: When in swap mode and the user has a connected account.
- Advanced Mode: When in advanced mode and advanced orders are unlocked.
- Limit Order Mode: When in limit order mode and limit orders are unlocked.

Also updated the button style:

<img width="697" alt="Screenshot 2024-03-11 at 16 27 51" src="https://github.com/cowprotocol/cowswap/assets/31534717/a32c8f5d-cd10-4da3-983b-d2b95dc14348">
<img width="597" alt="Screenshot 2024-03-11 at 16 27 41" src="https://github.com/cowprotocol/cowswap/assets/31534717/0f796198-a29b-4c78-b517-aff1ddd9a8ba">

elena-zh reviewed (approved):
Loos good now, thank you!
